### PR TITLE
Mcgcc 301 jsbox

### PR DESCRIPTION
**Pull request by DevChima**

### CODE REVIEW

**Comment by DevChima** [comment]
Here's the flow for this work: 
https://rapidpro.qa.momconnect.co.za/flow/editor_next/0dcf9bf5-75f0-4de9-a6b4-07733f3dae77/

The change flows only got ready so I need to add those screens as well. 
https://rapidpro.qa.momconnect.co.za/flow/editor_next/5fb3fffa-4380-4e44-903d-db0ba0537102/

**Comment by DevChima** [comment]
Change flows:
Change supporter:
https://rapidpro.qa.momconnect.co.za/flow/editor_next/360d8107-b92e-4d57-b23f-247d826b8f91/

Supporter profile and change
https://rapidpro.qa.momconnect.co.za/flow/editor_next/5fb3fffa-4380-4e44-903d-db0ba0537102/#

File: src/ussd_mcgcc_rapidpro.js [hunk 1]
```diff
@@ -77,25 +158,1008 @@ go.app = function() {
                 });
         });
 
-        self.states.add("state_welcome", function(name) {
+        self.states.add("state_mother_supporter_suggested_state", function(name) {
+            return new EndState(name, {
+                next: "state_start",
+                text: $(
+                    "Your supporter hasn't registered yet. " +
+                    "Remind them to dial [used#] to complete registration."
```
Comment (erikh360) [inline src/ussd_mcgcc_rapidpro.js]: Is `[used#]` a placeholder or something?

File: src/ussd_mcgcc_rapidpro.js [hunk 2]
```diff
@@ -77,25 +158,1008 @@ go.app = function() {
...
+                "[2/5]" +
+                "\nWhat language would you like to get messages in?"),
+            error:$(
+                "Sorry, please try again. " +
+                "Reply with the number, e.g. 1."
+            ), 
+            accept_labels: true,
+            choices: [
+                new Choice("eng_ZA", $("English")),
+                new Choice("afrikaans", $("Afrikaans")),
```
Comment (erikh360) [inline src/ussd_mcgcc_rapidpro.js]: Code is different format than the others

File: src/ussd_mcgcc_rapidpro.js [hunk 3]
```diff
@@ -77,25 +158,1008 @@ go.app = function() {
...
+                "\nWhat language would you like to get msgs in?"),
+            error:$(
+                "Please try again. " +
+                "Reply with the no, e.g. 1."
+            ), 
+            accept_labels: true,
+            choices: [
+                new Choice("zul_ZA", $("Zulu")),
+                new Choice("xho", $("Xhosa")),
+                new Choice("afrikaans", $("Afrikaans")),
```
Comment (erikh360) [inline src/ussd_mcgcc_rapidpro.js]: same here

File: src/ussd_mcgcc_rapidpro.js [hunk 4]
```diff
@@ -77,25 +158,1008 @@ go.app = function() {
...
+        return new ChoiceState(name, {
+            question: $(
+                "What language would you like to get messages in?"),
+            error:$(
+                "Reply with the nr that matches your answer, e.g. 1."
+            ), 
+            accept_labels: true,
+            choices: [
+                new Choice("eng_ZA", $("English")),
+                new Choice("afrikaans", $("Afrikaans")),
```
Comment (erikh360) [inline src/ussd_mcgcc_rapidpro.js]: same here

File: src/ussd_mcgcc_rapidpro.js [hunk 5]
```diff
@@ -77,25 +158,1008 @@ go.app = function() {
...
+            question: $(
+                "What language would you like to get msgs in?"),
+            error:$(
+                "Reply with the nr that matches your answer, e.g. 1."
+            ), 
+            accept_labels: true,
+            choices: [
+                new Choice("zul_ZA", $("Zulu")),
+                new Choice("xho", $("Xhosa")),
+                new Choice("afrikaans", $("Afrikaans")),
```
Comment (erikh360) [inline src/ussd_mcgcc_rapidpro.js]: and here

**Review by erikh360 — COMMENTED** [review]
just a few comments, more like questions, I didn't go into too much detail because of the size.

happy for you to merge

File: src/ussd_mcgcc_rapidpro.js (hunk 1 above)
Comment (DevChima) [reply src/ussd_mcgcc_rapidpro.js]: Thanks for picking this up. It's an old placeholder I added when I started. I'll fix now (the rest are intentional)
 I'll merge after correcting this.

**Pull request merged**